techdocs/script/operations.py

```diff
@@ -66,16 +74,119 @@ def _create_operation(self, file, rule):
         else:
             return None
         return FilesystemOperation(
-                type=FilesystemOperation.TYPE_COPY,
-                source_abs=path.abspath(path.join(self.from_path, relative_src)),
-                destination_abs=path.abspath(
-                    path.join(
-                        self.to_path,
-                        relative_dst, 
-                    ),
+            type=FilesystemOperation.TYPE_COPY,
+            source_abs=path.abspath(path.join(self.from_path, relative_src)),
+            destination_abs=path.abspath(
+                path.join(
+                    self.to_path,
+                    ProjectDetailsReader(self.to_path, filesystem).doc_path(rule.config.project),
+                    relative_dst,
                 ),
+            ),
+        )
+
+
+class DeleteDetector:
+    def __init__(self, repo, index, from_path, to_path, child) -> None:
+        self.repo = repo
+        self.index = index
+        self.from_path = from_path
+        self.to_path = to_path
+        self.child = child
+
+    def detect(self, filesystem):
+        child_result = self.child.detect(filesystem)
+        child_result_files_to_be_copied = [
+            path.relpath(op.destination_abs, self.to_path)
+            for op in child_result
+            if op.type == FilesystemOperation.TYPE_COPY
+        ]
+        indexed_items = list([i for i in self.index.items if i.repo == self.repo])
+        for item in indexed_items:
+            if item.file not in child_result_files_to_be_copied:
+                child_result.append(
+                    FilesystemOperation(
+                        type=FilesystemOperation.TYPE_DELETE,
+                        source_abs=path.abspath(
+                            path.join(self.to_path, item.file),
+                        ),
+                        destination_abs=None,
+                    )
+                )
+                self.index.remove(item)
+
+        for item in child_result_files_to_be_copied:
+            self.index.add(IndexItem(item, self.repo))
+        return child_result
+
+
+class Index:
```

Comment:
```suggestion
class FileIndex:
```
maybe excessive, but it's more readable straight away to me, could be applied to other classes.